main_routers/mmd_router.py

```
@@ -91,80 +93,116 @@ def _ensure_mmd_directory(config_mgr) -> Path | None:
+    except UnicodeDecodeError:
+        pass
+
+    # 3. 核心修改：给探测器戴上“紧箍咒”，只允许中日韩常见编码，坚决不要泰文
+    ALLOWED_ENCODINGS = {'cp932', 'shift_jis', 'gbk', 'gb18030', 'gb2312', 'big5', 'euc-kr'}
+    
+    results = charset_normalizer.from_bytes(raw_bytes_blob)
+    for result in results:
+        if result.encoding and result.encoding.lower() in ALLOWED_ENCODINGS:
+            return result.encoding.lower()
+
+    # 4. 降级安全网：如果探测器还是懵了，按 MMD 圈子最常见的概率强行试错
```

Ruff (0.15.9)

[warning] Comment contains ambiguous `：` (FULLWIDTH COLON). Did you mean `:` (COLON)?

(RUF003)